src/main/java/com/ddudu/infrastructure/persistence/repository/goal/GoalQueryRepositoryImpl.java

```diff
@@ -45,4 +54,32 @@ public List<GoalEntity> findAllByUserAndPrivacyTypes(
         .fetch();
   }
 
+  @Override
+  public List<CompletedDduduNumberStatsResponse> collectCompletedDduduNumberStats(
+      UserEntity user, LocalDate from, LocalDate to
+  ) {
+    StringPath completedCountAlias = Expressions.stringPath("completedCount");
+    NumberExpression<Integer> sum = new CaseBuilder().when(
+            dduduEntity.status.eq(DduduStatus.COMPLETE))
+        .then(Expressions.numberTemplate(Integer.class, "1"))
+        .otherwise(Expressions.numberTemplate(Integer.class, "0"))
+        .sum()
+        .as("completedCount");
```

Comment:
👍🏻

Reply:
이거 진짜 개고생했습니다
여담인데 `.then(1).otherwise(0)`했는데 `getJavaType()`을 할 수 없다는 Hibernate 예외가 자꾸 터지더라고요.
그래서 리서치하다가 `.then(Expressions.asNumber(1)).otherwise(Expressions.asNumber(0))`으로 NumberExpression을 담아줘야 한다길래 안되더라고요.
그래서 또 리서치 하다가 `.then(Expressions.asNumber(1).castToNum(Integer.class)).otherwise(Expressions.asNumber(0).castToNum(Integer.class))`로 지정해줘야 한다고 스택오버플로에서 그래서 했었는데 이것도 똑같은 예외가 터지더라고요.
근데 또 누구는 CaseBuilder사용하면 as 안 쓰면 모른다해서 했었는데 그래도 안된더라고요.
진짜 하루종일 찾아보다가 알고보니 Hibernate가 6버전에서 case when then으로 뭔가 바꿔서 Expressions.numberTemplate아니면 안되는 버그가 있었던겁니다. 우여곡절 끝에..

Reply:
ㅋㅋ큐ㅠ말만 들어도 머리 아프네요.. 덕분에 좋은거 배우고 갑니당..👍🏻
Hibernate에서 저런 버그도 나오네용.. 얼른 FIX 되길🥹
트러블 슈팅하느라 넘 고생하셨습니다..!!!!✊🏻